tests/Functional/Action/Lti/GetUserAssignmentLtiLinkActionTest.php

```diff
@@ -112,6 +112,32 @@ public function testItReturns409IfAssignmentDoesNotHaveASuitableState(): void
         );
     }
 
+    public function testItReturns409IfAssignmentIsNotAvailable(): void
+    {
+        Carbon::setTestNow(Carbon::createFromDate(2022, 1, 1));
+
+        /** @var UserRepository $userRepository */
+        $userRepository = $this->getRepository(User::class);
+        $user = $userRepository->getByUsernameWithAssignments('user1');
+
+        $user->getLastAssignment()->setState(Assignment::STATE_COMPLETED);
+        $this->getEntityManager()->flush();
+
+        $this->logInAs($user, $this->kernelBrowser);
+
+        $this->kernelBrowser->request('GET', '/api/v1/assignments/1/lti-link');
+
+        $this->assertEquals(Response::HTTP_NOT_FOUND, $this->kernelBrowser->getResponse()->getStatusCode());
+
+        $decodedResponse = json_decode($this->kernelBrowser->getResponse()->getContent(), true);
+        $this->assertEquals(
+            "Assignment id '1' not found for user 'user1'.",
```

Comment:
The usual question, which is though not related to test (and possibly could be fixed in another PR/task):
should we really return 404s when user is trying to launch assignment which is yet unavailable, but he/she sees assignment in list?

Reply:
I do agree with you. But I think that it's not only the 404 that is wrong, the message is as well. The assignment is found, but we cannot generate an LTI link, so looks more like conflict to me. I'll fix it in another task that I'll create as it touches many endpoints.

Reply:
https://oat-sa.atlassian.net/browse/TDP-63

Reply:
I confirm this is indeed an issue, it should throw a 409 exception if we try to retrieve LTI link for existing, but unavailable assignments.